packages/react-components/react-utilities/src/hooks/useMotionPresence.ts

```diff
@@ -0,0 +1,281 @@
+   * This should be used to conditionally render the element.
+   */
+  shouldRender: boolean;
+
+  /**
+   * Whether the element is visible in the DOM.
+   * This is true when the element is already rendered and transitioning from being hidden to visible.
+   */
+  visible: boolean;
+
+  /**
+   * Whether the element is entering the DOM.
+   * This is true when the element is transitioning from not being rendered to being rendered.
+   */
+  entering: boolean;
+
+  /**
+   * Whether the element is exiting the DOM.
+   * This is true when the element is transitioning from being rendered to not being rendered.
+   */
+  exiting: boolean;
+
+  /**
+   * Whether the element is animating.
+   * This is true when the element is entering or exiting the DOM.
+   */
+  animating: boolean;
+};
+
+/**
+ * Events for useMotionPresence hook.
+ */
+export type UseMotionPresenceEvents = {
+  /**
+   * Callback for after the element enters the DOM.
+   */
+  onEntered?: () => void;
+
+  /**
+   * Callback for after the element exits the DOM.
+   */
+  onExited?: () => void;
+};
+
+/**
+ * Returns CSS styles of the given node.
+ * @param node - DOM node.
+ * @returns - CSS styles.
+ */
+const getElementComputedStyle = (node: HTMLElement): Partial<CSSStyleDeclaration> => {
+  if (node.nodeType !== 1) {
+    return {
+      getPropertyValue: () => '',
+    };
+  }
+
+  const window = node.ownerDocument?.defaultView;
+
+  return window!.getComputedStyle(node, null);
+};
+
+/**
+ * @internal
+ * Converts a CSS duration string to milliseconds.
+ *
+ * @param s - CSS duration string
+ * @returns Duration in milliseconds
+ */
+function toMs(s: string): number {
+  if (s.includes('auto')) {
+    return 0;
+  }
+
+  if (s.includes('ms')) {
+    return parseFloat(s);
+  }
+
+  return Number(s.slice(0, -1).replace(',', '.')) * 1000;
+}
+
+/**
+ * @internal
+ * Gets the maximum duration from a list of CSS durations.
+ *
+ * @param durations - List of CSS durations
+ * @returns Maximum duration
+ */
+const getMaxCSSDuration = (durations: string[]) => {
+  return Math.max(...durations.map(d => toMs(d.trim())));
+};
+
+/**
+ * @internal
+ * Gets the motion information for a given element.
+ *
+ * @param computedStyle - Computed style of the element
+ * @returns motion information
+ */
+const getMotionInfo = (computedStyle: CSSStyleDeclaration) => {
+  const getProp = (prop: string) => (computedStyle?.getPropertyValue(prop) || '').split(',');
+
+  const transitionDuration = getProp('transition-duration');
+  const transitionDelay = getProp('transition-delay');
+  const animationDuration = getProp('animation-duration');
+  const animationDelay = getProp('animation-delay');
+  const totalTransitionDuration = getMaxCSSDuration(transitionDuration) + getMaxCSSDuration(transitionDelay);
+  const totalAnimationDuration = getMaxCSSDuration(animationDuration) + getMaxCSSDuration(animationDelay);
+
+  const hasAnimation = totalAnimationDuration > 0;
```

Comment:
Doesn't seem like we need both `hasAnimation` and `hasTransition` based on usage. These could be collapsed into `hasMotion`.

Reply:
This is not needed anymore. It was removed.